### PR TITLE
Adding info about docker vs podman extensions

### PR DESCRIPTION
**Pull request by edeandrea**
Adding some additional info about why the need for separate Docker and podman extensions

### CODE REVIEW

**Comment by edeandrea** [comment]
@maxandersen how's this look?

File: docs/src/main/asciidoc/container-image.adoc [hunk 1]
```diff
@@ -131,6 +136,16 @@ To use this feature, add the following extension to your project.
 :add-extension-extensions: container-image-podman
 include::{includes}/devtools/extension-add.adoc[]
 
+[TIP]
+.When to use Docker vs Podman extension
+====
+The <<#docker,Docker extension>> is and has always been backwards-compatible with Podman because Podman exposes a https://podman.io/docs/installation[Docker-compatible API]. You can build container images with Podman using the Docker extension (see the https://quarkus.io/guides/podman[Using Podman with Quarkus guide]).
+
+You need to use either the Docker or Podman specific extension when you need to do things that are specific to either Docker or Podman.
```
Comment (gastaldi) [inline docs/src/main/asciidoc/container-image.adoc:144]:
```suggestion
You need to use either the `quarkus-container-image-docker` or `quarkus-container-image-podman` extension when you need to do things specific to either Docker or Podman.
```

Comment (gastaldi) [reply docs/src/main/asciidoc/container-image.adoc:144]: I think specifying the extension names is clearer than stating "Docker or Podman specific extension"

Comment (edeandrea) [reply docs/src/main/asciidoc/container-image.adoc]: Agreed. I committed your suggestion.

Comment (gastaldi) [reply docs/src/main/asciidoc/container-image.adoc]: That was fast, I added some minor wording improvements to that sentence as well, see if that makes sense and please squash the commits before merging 😉

Comment (edeandrea) [reply docs/src/main/asciidoc/container-image.adoc]: I think what you wrote is fine. Can't the commits be squashed when merging? Squash & commit vs Merge & commit?

Comment (gastaldi) [reply docs/src/main/asciidoc/container-image.adoc]: image

Comment (edeandrea) [reply docs/src/main/asciidoc/container-image.adoc]: I was going to ask "Why not?", but I'll save that for another day when I have time for a longer debate :) I'll go ahead and rebase now...

Comment (edeandrea) [reply docs/src/main/asciidoc/container-image.adoc]: Rebase done

Comment (edeandrea) [reply docs/src/main/asciidoc/container-image.adoc]: Weird when I clicked on "Commit suggestion" it didn't commit the whole suggestion. Just fixed & rebased

Comment (gastaldi) [reply docs/src/main/asciidoc/container-image.adoc]: It's because of how we backport PRs, squashing and merging would make the commit untraceable

**Review by gastaldi — APPROVED** [review]

**Comment by edeandrea** [comment]
Any chance getting this merged?

**Comment by gastaldi** [comment]
Thanks!

**Pull request merged**